### PR TITLE
Add header files to extra-source-files in cabal file

### PR DESCRIPTION
**Pull request by newhoggy**
description
-----------
Add header files to `extra-source-files`.  This is required to build downstream projects with `cabal-3.4`, and is also necessary for `cabal install` to work in downstream projects for earlier cabal versions.

checklist
---------

- [x] compiles (`cabal v2-build` or `stack build`)
- [x] tests run successfully (`cabal v2-test` or `stack test`)
- [ ] documentation added
- [ ] link to an issue
- [ ] add milestone (the current sprint)

### CODE REVIEW

File: iohk-monitoring/iohk-monitoring.cabal [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
+license:              Apache-2.0
+license-files:        LICENSE, NOTICE
+author:               Alexander Diemand, Andreas Triantafyllos
+maintainer:           operations@iohk.io
+copyright:            2018 IOHK
+category:             Benchmarking
+build-type:           Simple
+extra-source-files:   README.md
+                      src/Cardano/BM/Counters/os-support-darwin.h
+                      src/Cardano/BM/Counters/os-support-win.h
```
Comment (CodiePP) [inline iohk-monitoring/iohk-monitoring.cabal:14]: what is the reason for including these files here?
why did you not include the .c files?

Comment (newhoggy) [reply iohk-monitoring/iohk-monitoring.cabal:14]: It was to address this error:

```
Darwin.hsc:30:10: fatal error: 'os-support-darwin.h' file not found
#include "os-support-darwin.h"
         ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
compiling dist/build/Cardano/BM/Counters/Darwin_hsc_make.c failed (exit code 1)
command was: /usr/bin/gcc -c dist/build/Cardano/BM/Counters/Darwin_hsc_make.c -o dist/build/Cardano/BM/Counters/Darwin_hsc_make.o -fno-stack-protector -fno-stack-protector -D__GLASGOW_HASKELL__=806 -Ddarwin_BUILD_OS=1 -Dx86_64_BUILD_ARCH=1 -Ddarwin_HOST_OS=1 -Dx86_64_HOST_ARCH=1 -Isrc/Cardano/BM/Counters/ -Idist/build/src/Cardano/BM/Counters/ -DENABLE_OBSERVABLES -Idist/build/autogen -Idist/build/global-autogen -include dist/build/autogen/cabal_macros.h -I/Users/jky/.cabal/store/ghc-8.6.5/ntwrk-3.1.1.1-58eabeff/lib//include -I/Users/jky/.cabal/store/ghc-8.6.5/vctr-lgrthms-0.8.0.3-028b099e/lib//include -I/Users/jky/.cabal/store/ghc-8.6.5/vctr-0.12.1.2-477dab05/lib//include -I/Users/jky/.ghcup/ghc/8.6.5/lib/ghc-8.6.5/unix-2.7.2.2/include -I/Users/jky/.ghcup/ghc/8.6.5/lib/ghc-8.6.5/time-1.8.0.2/include -I/Users/jky/.cabal/store/ghc-8.6.5/prmtv-0.7.1.0-0a78c306/lib//include -I/Users/jky/.ghcup/ghc/8.6.5/lib/ghc-8.6.5/bytestring-0.10.8.2/include -I/Users/jky/.ghcup/ghc/8.6.5/lib/ghc-8.6.5/base-4.12.0.0/include -I/Users/jky/.ghcup/ghc/8.6.5/lib/ghc-8.6.5/integer-gmp-1.0.2.0/include -I/Users/jky/.ghcup/ghc/8.6.5/lib/ghc-8.6.5/include -I/Users/jky/.ghcup/ghc/8.6.5/lib/ghc-8.6.5/include/
cabal: Failed to build iohk-monitoring-0.1.10.1 (which is required by
exe:cardano-node-chairman from cardano-node-chairman-1.19.1). See the build
log above for details.
```

The `.c` files are already mentioned in the cabal file so those are probably fine.

Comment (erikd) [reply iohk-monitoring/iohk-monitoring.cabal:14]: I really do not think this is the right solution.

**Review by erikd — APPROVED** [review]

File: iohk-monitoring/iohk-monitoring.cabal (hunk 1 above)
Comment (erikd) [reply iohk-monitoring/iohk-monitoring.cabal:14]: @CodiePP is right to question why the `.c` files are not also added. Does `cabal sdist` work in this repo?

Comment (erikd) [reply iohk-monitoring/iohk-monitoring.cabal:14]: Just make sure that `cabal sdist` works correctly.

Comment (newhoggy) [reply iohk-monitoring/iohk-monitoring.cabal:14]: The below example shows `c` files are handled correctly as well.

```
➜  iohk-monitoring-framework git:(extra-source-files) cabal sdist iohk-monitoring
Cloning into '/Users/jky/wrk/iohk/iohk-monitoring-framework/dist-newstyle/src/ouroboros_-e7dffa0d85e2839'...
remote: Enumerating objects: 674, done.
remote: Counting objects: 100% (674/674), done.
remote: Compressing objects: 100% (544/544), done.
remote: Total 99496 (delta 411), reused 211 (delta 85), pack-reused 98822
Receiving objects: 100% (99496/99496), 98.50 MiB | 9.24 MiB/s, done.
Resolving deltas: 100% (68515/68515), done.
HEAD is now at 3b92f6e93 Merge #2222
Wrote tarball sdist to
/Users/jky/wrk/iohk/iohk-monitoring-framework/dist-newstyle/sdist/iohk-monitoring-0.1.10.1.tar.gz
➜  iohk-monitoring-framework git:(extra-source-files) tar tvf /Users/jky/wrk/iohk/iohk-monitoring-framework/dist-newstyle/sdist/iohk-monitoring-0.1.10.1.tar.gz | grep '\.\(h\|c\)$'
-rw-r--r--  0 0      0        9408  9 Sep  2001 iohk-monitoring-0.1.10.1/src/Cardano/BM/Counters/os-support-darwin.c
-rwxr-xr-x  0 0      0        1020  9 Sep  2001 iohk-monitoring-0.1.10.1/src/Cardano/BM/Counters/os-support-darwin.h
-rw-r--r--  0 0      0        2943  9 Sep  2001 iohk-monitoring-0.1.10.1/src/Cardano/BM/Counters/os-support-win.c
-rwxr-xr-x  0 0      0         429  9 Sep  2001 iohk-monitoring-0.1.10.1/src/Cardano/BM/Counters/os-support-win.h
```

**Review by CodiePP — APPROVED** [review]

**Pull request merged**